mitodl/mit-open
# topic detail banner / subtopic logic revisions

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5361

### Description (What does it do?)
This PR moves some things around relating to topic channel pages:
 - Subtopics moved from channel page body to the banner
 - New `ChannelPageTemplate` component created to give topic channels their own template
 - Clicking on Subtopics to load their channel pages now shows "Related Topics" aka their siblings as Subtopics of the parent topic
 - Action buttons moved below the channel title
 - Various style changes

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/c07d7ca9-f6b3-4ad7-88ef-7a09a8cb2063)
![image](https://github.com/user-attachments/assets/1404546d-8357-4f3c-a928-429febc496e8)

### How can this be tested?
 - Spin up this branch of `mit-learn`
 - Ensure that you have enough data backpopulated to have some topics and courses in those topics
 - Visit http://localhost:8062/topics
 - Click one of the top level topics that contain subtopics to visit its channel page
 - Verify that the subtopics display in the manner described above and on the issue at desktop and mobile resolutions
 - Click one of the subtopic chips
 - Ensure that the "Subtopics" title above the chips now says "Related Topics"
 - Verify that you can click the 3rd tier breadcrumb and go back to the parent topic
